New post: Database (#60)
FalixNodes-Software/help-center: _posts/falix/general/2021-09-20-database.md

---
layout: post
title:  "Database"
categories: Falix
tags: fxgeneral
permalink: /falix/general/database/
---

# Guide On How To Use Databases
## What Are Databases?

> Databases are an organized collection of information. In Falix we support MariaDB databases.

Reviewer: ```suggestion
Databases are an organized collection of information. 
The database our panel support is [MariaDB](https://mariadb.org).
```

---

Maybe also provide some information on Maria as well.

---
layout: post
title:  "Database"
categories: Falix
tags: fxgeneral
permalink: /falix/general/database/
---

# Guide On How To Use Databases
## What Are Databases?
Databases are an organized collection of information. 
The database our panel supports is [MariaDB](https://mariadb.org).
MariaDB is a very popular open source database which was developed by the original MySQL developers.

## What Are Databases Useful For? 

> They are needed by some plugins/bots to store data. If you are using a plugin, it usually states whether it needs a database or not in its wiki page.

Reviewer: ```suggestion
Some server plugins, bots, or systems may require a database to store information. Minecraft plugins will usually state whether they need/support an SQL database.
```

# Getting Started
## Step 1: Allocating A Database To Your Server

1. Go to the [Client Panel](https://client.falixnodes.net).
2. Scroll down and click on "Add Database".
3. The page will reload, you should see a success message above the list of your servers.
![image](../../../assets/images/posts/falix/database/client-page-step-1.png)

## Step 2: Creating The Database

1. Go to the [Game Panel](https://panel.falixnodes.net).
2. Click on your server.
3. Click on "Databases".
4. Click on "New Database", it should be a big blue button at the bottom right of the page.
![image](../../../assets/images/posts/falix/database/new-database.png)
5. You should see a popup appear, enter in any name you like in the "Database Name" field.
6. For the second field you can enter in the node's IP, or leave it as `%`.
7. Once you have filled in the fields, click on "Create Database".
![image](../../../assets/images/posts/falix/database/create-database.png)
8. Your database should be created, click on the eye icon to view your database credentials.

## Step 3: Viewing And Editing The Contents Of Your Database

> ### What is Phpmyadmin?
> Phpmyadmin is a free tool used to access and edit databases.
> 
> 1. Copy all your database credentials, this is needed to access the database.
> 2. Go to [Phpmyadmin](https://phpmyadmin.co/).
> 3. Insert your database credentials into the fields provided, follow the table below.
> 
> | Database credentials                      |           =       | Login credentials                                     |
> |-------------------------------------------|-------------------|-------------------------------------------------------|
> | Endpoint                                  |           =       | Server                                                |
> |-------------------------------------------|-------------------|-------------------------------------------------------|
> | Username                                  |           =       | Username                                              |
> |-------------------------------------------|-------------------|-------------------------------------------------------|
> | Password                                  |           =       | Password                                              |
> 
> It should look like this:
> ![image](../../../assets/images/posts/falix/database/accessing-database.png)
> 4. Click on "Go".
> 5. You should be logged in.
> 
> > Remember to put `%` in the "Connections from" field in the database's settings before attempting to access it.

Reviewer: PHPMyAdmin is not the only way, nor should it be the recommend way to access/manage a database.
```suggestion
The database we support is [MariaDB](https://mariadb.org). 
To connect, simply use the server node IP and the credentials given.

If you'd like to view or edit information inside of the database from a website, you may use [phpMyAdmin](https://phpmyadmin.co).
```

...Or something...

Author: But some users dont know how to login, what should we do about that?

Reviewer: Not our problem. We are not phpMyAdmin support.

Author: > simply use the server node IP and the credentials given.

Are you talking about the server IP or the database endpoint?

Reviewer: > Are you talking about the server IP or the database endpoint?

Whatever IP you use to connect to the database.
It does sound confusing to me, so please feel free to change the wording.
